packages/sogrim-app/src/components/App/UserApp.tsx

```diff
@@ -1,31 +1,34 @@
-import { CssBaseline, ThemeProvider } from "@mui/material";
+import { CssBaseline, useTheme } from "@mui/material";
 import { observer } from "mobx-react-lite";
-import { useEffect, useMemo, useState } from "react";
+import React from "react";
+import { useEffect } from "react";
 import useUserState from "../../hooks/apiHooks/useUserState";
 import { useAuth } from "../../hooks/useAuth";
 import { useStore } from "../../hooks/useStore";
-import { DARK_MODE_THEME, LIGHT_MODE_THEME } from "../../themes/constants";
-import { getAppTheme } from "../../themes/theme";
+import { DARK_MODE_THEME } from "../../themes/constants";
 import { Layout } from "../Layout/Layout";
+import { ColorModeContext } from "./App";
 
 const UserAppComp: React.FC = () => {
-  const [mode] = useState<typeof LIGHT_MODE_THEME | typeof DARK_MODE_THEME>(
-    LIGHT_MODE_THEME
-  );
-  const theme = useMemo(() => getAppTheme(mode), [mode]);
-
   const { userAuthToken } = useAuth();
   const { data, isLoading, isError, error } = useUserState(userAuthToken);
   const {
     dataStore: { updateStoreUserDetails, updateStoreUserSettings },
     uiStore: { computeUserRegistrationState, userRegistrationState },
   } = useStore();
 
+  const theme = useTheme();
+  const colorMode = React.useContext(ColorModeContext);
+
   useEffect(() => {
     if (!isLoading && data) {
       updateStoreUserDetails(data.details);
       updateStoreUserSettings(data.settings);
+      if (data.settings.dark_mode && theme.palette.mode !== DARK_MODE_THEME) {
+        colorMode.toggleColorMode();
+      }
```

Comment:
I think there might be a bug here.
What will happen in the following scenario:
The user changes in his device to dark mode. Now the value of `data.settings.dark_mode` is `true`.
Next, he is connecting from a different device, and changes back to `light mode`.
IMHO there will be a limbo state, where his original device will still be dark mode and his second device light mode.

Reply:
Ok I found a way to test it locally and it seems I am wrong :).
Closing the comment.